astropy/io/fits/hdu/hdulist.py

```diff
@@ -1079,7 +1079,7 @@ def _readfrom(cls, fileobj=None, data=None, mode=None,
 
             raise OSError('Empty or corrupt FITS file')
 
-        if not lazy_load_hdus:
+        if not lazy_load_hdus or kwargs.get('checksum') is True:
```

Comment:
Why the `is True`? Wouldn't `kwargs.get('checksum)` be sufficient?

Reply:
Because it can set to 'remove', in which case the checksums are not verified.

Reply:
Ah, right. Still weird 🤔